WordPress/Classes/ViewRelated/Post/AztecAttachmentViewController.swift

```diff
@@ -0,0 +1,195 @@
+            "DefaultValue": alignment,
+            "Title": NSLocalizedString("Alignment", comment:"Title of the screen for choosing an image's alignment."),
+            "Titles": titles,
+            "Values": values,
+            "CurrentValue": currentValue
+        ]
+
+        guard let vc = SettingsSelectionViewController(dictionary: dict) else {
+            return
+        }
+
+        vc.onItemSelected = { (status: Any) in
+            if let newAlignment = status as? TextAttachment.Alignment {
+                self.alignment = newAlignment
+            }
+            vc.dismiss()
+            self.tableView.reloadData()
+        }
+
+        self.navigationController?.pushViewController(vc, animated: true)
+    }
+
+    func displaySizeSelector(row: ImmuTableRow) {
+        let values = sizeTitles.map { (key: TextAttachment.Size, value: String) -> TextAttachment.Size in
+            return key
+        }
+
+        let titles = values.map { (value: TextAttachment.Size) -> String in
+            return sizeTitles[value]!
+        }
+
+        let currentValue = size
+
+        let dict: [String: Any] = [
+            "DefaultValue": size,
+            "Title": NSLocalizedString("Image Size", comment: "Title of the screen for choosing an image's size."),
+            "Titles": titles,
+            "Values": values,
+            "CurrentValue": currentValue
+        ]
+
+        guard let vc = SettingsSelectionViewController(dictionary: dict) else {
+            return
+        }
+        vc.onItemSelected = { (status: Any) in
+            // do interesting work here... like updating the value of image meta.
+            if let newSize = status as? TextAttachment.Size {
+                self.size = newSize
+            }
+            vc.dismiss()
+            self.tableView.reloadData()
+        }
+
+        self.navigationController?.pushViewController(vc, animated: true)
+    }
+
+    // MARK: - Helper methods
+
+    func handleCancelButtonTapped(sender: UIBarButtonItem) {
+        dismiss(animated: true, completion: nil)
+    }
+
+    func handleDoneButtonTapped(sender: UIBarButtonItem) {
+        if let attachment = self.attachment {
+            attachment.alignment = alignment
+            attachment.size = size
+            delegate?.aztecAttachmentViewController(self, changedAttachment: attachment)
+        }
+        dismiss(animated: true, completion: nil)
+    }
+
+    private let alignTitles: [TextAttachment.Alignment: String] = [
+        .left: NSLocalizedString("Left", comment: "Left alignment for an image. Should be the same as in core WP."),
+        .center: NSLocalizedString("Center", comment: "Center alignment for an image. Should be the same as in core WP."),
+        .right: NSLocalizedString("Right", comment: "Right alignment for an image. Should be the same as in core WP."),
+        .none: NSLocalizedString("None", comment: "No alignment for an image (default). Should be the same as in core WP.")
+    ]
```

Comment:
**No need to change anything / just a comment to consider**

I used to hack implementations exactly like this one, but changed slightly. If, tomorrow, we were to add another Alignment case, we wouldn't get a warning for a missing title, and we would just notice that after crashing the app.

For that reason, nowadays i implement similar functionality using this pattern:

```
extension TextAttachment.Alignment {
    var title {
       switch self {
          case .left: 
              return NSLocalizedString(...)
          // ...
       }
   }
}
```

**And as a bonus:** you get to nuke the ImplicitlyUnwrappedOptionals in snippets such as... `alignTitles[.left]!`